src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1477,6 +1478,26 @@ public async Task WaitForExitAsync(CancellationToken cancellationToken = default
             finally
             {
                 Exited -= handler;
+
+                await WaitUntilOutputEOF().ConfigureAwait(false);
+            }
+
+            // Wait until output streams have been drained
+            async ValueTask WaitUntilOutputEOF()
+            {
+                // if we have a hard timeout, we cannot wait for the streams
```

Comment:
What is the timeout being referred to here?  WaitForExitAsync takes a CancellationToken; this is in contrast to WaitForExit taking a milliseconds timeout.

Reply:
Rather than this CanBeCanceled check, did you try passing the cancellation token into WaitUntilEOFAsync?  Is there a downside to doing so, e.g. does it somehow lead to inconsistent states?

Reply:
No real downside, other than slightly different behaviour compared to the sync method. I've updated the code so that the cancellation token is passed along to the wait methods.